### ISSUE TITLE
Short options support "=" assignment operator in complete builtin

### ISSUE THREAD

**Issue by EmilySeville7cfg** [fish-shell/fish-shell#8466]
Fish version: fish, version 3.3.1-622-ge06671512 (built from source)
OS version: Linux Mint 20.2 Cinnamon

```fish
complete --command : --short-option s --arguments 'arg1 arg2'
```

![image](https://user-images.githubusercontent.com/42812113/142745497-bd1d67a9-0118-4067-93a2-9e3c66adb201.png)

Note that when using `=` no `arg1 arg2` appear in suggestions.

**Comment by krobelus** [comment]
Is there a program that parses short-option arguments like that? With GNU getopt, an `-s=arg` is interpreted as an option with the argument `=arg`

**Comment by page-down** [comment]
python for example

```python
import argparse
parser = argparse.ArgumentParser()
parser.add_argument('-f')
args = parser.parse_args()
print(vars(args))
```

```
python3 test.py -f=test
{'f': 'test'}
```

There are also many programs that support this kind of parsing.
For example, ripgrep `rg` .

**Comment by faho** [comment]
Since it's not available in getopt, this shouldn't be offered by default - it would break getopt-using programs, and those are rather common. I don't think an option to specify the dialect is worth it because you can always pass it as `-ftest` or `-f test`.

What we should do is remove the special "=" handling in this case entirely - parse it like getopt does, as an argument "=arg".

**Comment by page-down** [comment]
I agree that by default `complete --short-option s` should resolve to `=arg1` when supplied with `-s=arg1`.

However, when a program expects `-s=arg1` to get `arg1`, `complete` should be allowed to be supported (when it does not have a negative impact).

Users should not be required to change the way they enter arguments, or require programs to follow strict rules. (The implication is that those programs that do not conform to this strict rule will never be supported.)
